packages/amplication-server/prisma/migrations/20211230120749_prisma_v3_renaming/migration.sql

```diff
@@ -0,0 +1,71 @@
+-- AlterTable
+ALTER TABLE "Workspace" RENAME CONSTRAINT "Organization_pkey" TO "Workspace_pkey";
+
+-- RenameIndex
+ALTER INDEX "Account.currentUserId_unique" RENAME TO "Account_currentUserId_key";
+
+-- RenameIndex
+ALTER INDEX "Account.email_unique" RENAME TO "Account_email_key";
+
+-- RenameIndex
+ALTER INDEX "ApiToken.userId_name_unique" RENAME TO "ApiToken_userId_name_key";
+
+-- RenameIndex
+ALTER INDEX "App.workspaceId_name_unique" RENAME TO "App_workspaceId_name_key";
+
+-- RenameIndex
+ALTER INDEX "AppRole.appId_displayName_unique" RENAME TO "AppRole_appId_displayName_key";
+
+-- RenameIndex
+ALTER INDEX "AppRole.appId_name_unique" RENAME TO "AppRole_appId_name_key";
+
+-- RenameIndex
+ALTER INDEX "Block.appId_displayName_unique" RENAME TO "Block_appId_displayName_key";
+
+-- RenameIndex
+ALTER INDEX "BlockVersion.blockId_versionNumber_unique" RENAME TO "BlockVersion_blockId_versionNumber_key";
+
+-- RenameIndex
+ALTER INDEX "Build.appId_version_unique" RENAME TO "Build_appId_version_key";
+
+-- RenameIndex
+ALTER INDEX "Entity.appId_displayName_unique" RENAME TO "Entity_appId_displayName_key";
+
+-- RenameIndex
+ALTER INDEX "Entity.appId_name_unique" RENAME TO "Entity_appId_name_key";
+
+-- RenameIndex
+ALTER INDEX "Entity.appId_pluralDisplayName_unique" RENAME TO "Entity_appId_pluralDisplayName_key";
+
+-- RenameIndex
+ALTER INDEX "EntityField.entityVersionId_displayName_unique" RENAME TO "EntityField_entityVersionId_displayName_key";
+
+-- RenameIndex
+ALTER INDEX "EntityField.entityVersionId_name_unique" RENAME TO "EntityField_entityVersionId_name_key";
+
+-- RenameIndex
+ALTER INDEX "EntityField.entityVersionId_permanentId_unique" RENAME TO "EntityField_entityVersionId_permanentId_key";
+
+-- RenameIndex
+ALTER INDEX "EntityPermission.entityVersionId_action_unique" RENAME TO "EntityPermission_entityVersionId_action_key";
+
+-- RenameIndex
+ALTER INDEX "EntityPermissionField.permissionId_fieldPermanentId_unique" RENAME TO "EntityPermissionField_permissionId_fieldPermanentId_key";
+
+-- RenameIndex
+ALTER INDEX "EntityPermissionRole.entityVersionId_action_appRoleId_unique" RENAME TO "EntityPermissionRole_entityVersionId_action_appRoleId_key";
+
+-- RenameIndex
+ALTER INDEX "EntityVersion.entityId_versionNumber_unique" RENAME TO "EntityVersion_entityId_versionNumber_key";
+
+-- RenameIndex
+ALTER INDEX "Environment.appId_name_unique" RENAME TO "Environment_appId_name_key";
+
+-- RenameIndex
+ALTER INDEX "User.accountId_workspaceId_unique" RENAME TO "User_accountId_workspaceId_key";
+
+-- RenameIndex
+ALTER INDEX "UserRole.userId_role_unique" RENAME TO "UserRole_userId_role_key";
+
+-- RenameIndex
+ALTER INDEX "Workspace.name_unique" RENAME TO "Workspace_name_key";
```

Comment:
Why all these changes are needed?

Reply:
**Option 1: I want to maintain my existing constraint and index names**
If you want to keep your database unchanged and keep the existing names for constraints and indexes you need to pull them into your schema so Prisma is aware of them.
**Option 2: I want to use Prisma's default constraint and index names**
If you want to keep your Prisma Schema clean and if you have no reasons preventing you from renaming constraints and indexes in your database, then you can create a migration to update the names.

Reply:
The standard names for indexes in PostgreSQL are:

`{tablename}_{columnname(s)}_{suffix}`

where the suffix is one of the following:

`pkey` for a Primary Key constraint
`key`f or a Unique constraint
`excl` for an Exclusion constraint
`idx` for any other kind of index
`fkey` for a Foreign key
`check` for a Check constraint
Standard suffix for sequences is

`seq` for all sequences

from: 
- [stackoverflow](https://stackoverflow.com/questions/4107915/postgresql-default-constraint-names/4108266#4108266)
- [postgresql manual](https://www.postgresql.org/docs/current/sql-altertable.html)

Reply:
This can be removed for now

Reply:
Just wondering what made us do it now, no reason to pull it out.
IIUC this was generated by the new Prisma version? 

Anyway, it would have been better to use a separate PR with explanation for history tracking.

My two cents, your call

Reply:
Yes, this is generated by the new Prisma version. This change can happen in any time in the future, so I don't see any reason to do it now